### PR TITLE
MAYA-107087: Don't allow parenting to a Gprim.

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
We need to block the parenting of one gprim to another. Pixar lists this behavior as "strongly discouraged", but we are also seeing evidence that this behavior is not even fully supported through hydra.

https://github.com/Autodesk/maya-usd/issues/708#issuecomment-695012836

### CODE REVIEW

File: lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp [hunk 1]
```diff
@@ -59,6 +60,17 @@ UsdUndoInsertChildCommand::UsdUndoInsertChildCommand(const UsdSceneItem::Ptr& pa
...
+    // Don't allow parenting to a Gprim.
+    // USD strongly discourages parenting of one gprim to another.
+    // https://graphics.pixar.com/usd/docs/USD-Glossary.html#USDGlossary-Gprim
+    if (parentPrim.IsA<UsdGeomGprim>()) {
+        std::string err = TfStringPrintf("Parenting geometric prim [%s] under geometric prim [%s] is not allowed "
+                                         "Please parent geometric prims under separate XForms and reparent between XForms.",
+                                         childPrim.GetName().GetString().c_str(),
+                                         parentPrim.GetName().GetString().c_str());
+        throw std::runtime_error(err.c_str());
+    }
```
Comment (HamedSabri-adsk) [inline lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp:72]: We thrown an error if one tries to parent a Gprim under another.

File: test/lib/ufe/testParentCmd.py [hunk 2]
```diff
@@ -470,3 +473,14 @@ def checkUnparent(done):
...
+    def testParentingToGPrim(self):
+        '''Parenting an object to UsdGeomGprim object is not allowed'''
+
+        # open tree scene
+        mayaUtils.openTreeScene()
+
+        with self.assertRaises(RuntimeError):
+            cmds.parent("|Tree_usd|Tree_usdShape,/TreeBase/trunk",
+                        "|Tree_usd|Tree_usdShape,/TreeBase/leavesXform/leaves")
+
```
Comment (HamedSabri-adsk) [inline test/lib/ufe/testParentCmd.py:486]: Added a unit test to back the Gprim parenting restriction.

Comment (HamedSabri-adsk) [reply test/lib/ufe/testParentCmd.py:486]: @ppt-adsk  To make testParentRelative, testParentAbsolute working under the new rule, I created a top level Xforms for cube, cylinder, sphere in "simpleSceneTRS.usda" and made sure the Translate/Rotate/Scale and Order Ops matches the underlying meshes. Github won't show the diff results. You need to use your diff-editor locally to see the changes.

**Review by ppt-adsk — APPROVED** [review]

File: test/lib/ufe/testSelection.py [hunk 3]
```diff
@@ -66,13 +66,14 @@ def runTestSelection(self, selectCmd):
...
+        mayaNames = ["pCube1", "pCylinder1", "pSphere1"]
         usdPaths = []
-        for n in ["/" + o for o in names]:
+        for n in ["/" + o for o in ufeNames]:
             usdPaths.append(ufe.Path(
                 [shapeSegment, usdUtils.createUfePathSegment(n)]))
         mayaPaths = []
-        for n in ["|world|" + o for o in names]:
+        for n in ["|world|" + o for o in mayaNames]:
             mayaPaths.append(ufe.Path(mayaUtils.createUfePathSegment(n)))
```
Comment (HamedSabri-adsk) [inline test/lib/ufe/testSelection.py:77]: @ppt-adsk Adopt to the recent changes in simpleSceneTRS:  Separated "maya" and "ufe" names list.

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**